Lesson12.py

```diff
@@ -0,0 +1,83 @@
+from functools import reduce
+
+"""
+Task 1
+Write a Python program to detect the number of local variables declared in a function.
+"""
+
+
+def func():
+    """
+    This is testing function
+    :return:
+    """
+    a = 'sdfsf'
+    bsdfsf = 2323
+    t = 85
+    ut = 52.25
+
+
+def count(fun):
+    sum_local_var_fun = fun.__code__.co_nlocals
+    return sum_local_var_fun
+
+print(count(func))  # output
+
+
+"""
+Task 2
+
+Write a Python program to access a function inside a function (Tips: use function, which returns another function)
+"""
+
+
+def first_func():
+    print('This is first function')
+
+
+def second_func():
+    print('This is second function')
+    return first_func
+
+
+return_func = second_func()
+
+return_func()
+
+
+"""
+Task 3
+
+Write a function called `choose_func` which takes a list of nums and 2 callback functions. 
+If all nums inside the list are positive, execute the first function on that list and return the result of it.
+Otherwise, return the result of the second one
+"""
+
+
+def choose_func(nums: list, func1, func2):
+    if reduce(lambda x, y: abs(x + y), nums) == reduce(lambda x, y: x + y, nums):
```

Comment:
To complex expression. I'd suggest something like:
`all(map(lambda x: x >= 0, nums))`
or
`len([i for i in nums if i < 0])`
or
`len(tuple(filter(lambda x: x < 0, a)))`
and so on...

Or just to use regular for-loop above. :)

Reply:
I don’t even know why I made such a long record) I changed it to one of the proposed options - it works the same way)) updated in the Lesson12 brunch